File: backend/src/main/java/com/hospital/hms/service/DepartmentService.java
package com.hospital.hms.service;

import com.hospital.hms.dto.request.DepartmentRequest;
import com.hospital.hms.dto.response.DepartmentResponse;

import java.util.List;

public interface DepartmentService {

    DepartmentResponse create(DepartmentRequest request);

    DepartmentResponse update(Long id, DepartmentRequest request);

    DepartmentResponse getById(Long id);

    List<DepartmentResponse> getAll();

    void delete(Long id);
}

File: frontend/src/services/departmentService.js
import apiClient from './apiClient';

const departmentService = {
  async getAll() {
    const { data } = await apiClient.get('/departments');
    return data.data;
  },
  async getById(id) {
    const { data } = await apiClient.get(`/departments/${id}`);
    return data.data;
  },
  async create(payload) {
    const { data } = await apiClient.post('/departments', payload);
    return data.data;
  },
  async update(id, payload) {
    const { data } = await apiClient.put(`/departments/${id}`, payload);
    return data.data;
  },
  async remove(id) {
    await apiClient.delete(`/departments/${id}`);
  },
};

export default departmentService;
